metadeploy/api/admin.py

```diff
@@ -156,8 +159,27 @@ class PlanTemplateAdmin(MetadeployTranslatableAdmin):
     pass
 
 
+class PlanAdminForm(forms.ModelForm):
+    class Meta:
+        model = Plan
+        fields = "__all__"
+
+    def clean(self):
+        cleaned_data = super().clean()
+        visible_to = cleaned_data.get("visible_to")
+        supported_orgs = cleaned_data.get("supported_orgs")
+        if visible_to and supported_orgs != SUPPORTED_ORG_TYPES.Persistent:
+            raise forms.ValidationError(
+                _(
+                    'Restricted plans (with a "visible to" AllowedList) can only support persistent org types.'
+                )
+            )
+        return cleaned_data
```

Comment:
Is this the right place for this validation? It's performed for changes through the admin UI, but not via the admin REST API.

Reply:
Maybe we can raise `ValidationError` on the model's `clean` method (which would work out of the box in the admin I think) and make sure it's called by the viewset's create and update methods, where it's translated to something DRF can re-surface.

Reply:
@davisagli (cc @jerivas) I moved this to the model `clean` method, which works similarly when changed via the admin.

The `PlanViewSet` inherits from `ReadOnlyModelViewSet`, so there is no support practically for any create/update actions via the DRF API as far as I can tell. So I did **not** add validation to the `PlanSerializer`, but I'm open to doing that if you'd prefer.

Reply:
@jgerigmeyer you're thinking about the API that supports the frontend, but there is also the separate admin API that we use for publishing new plan versions: https://github.com/oddbird/MetaDeploy/blob/main/metadeploy/adminapi/api.py#L50

Reply:
@davisagli Ah, makes sense. I can add the same validation to the serializer(s) then.

Reply:
@davisagli Added validation to both serializers. Would you prefer a shared helper fn rather than duplicated?